### PR TITLE
Implement Intel leaf 0x1F (Extended Topology Information V2)

### PR DESCRIPTION
**Pull request by semiviral**
Simple implementation for the 2nd iteration of the extended topology leaf.

### CODE REVIEW

File: src/tests/xeon_gold_6252.rs [hunk 1]
```diff
@@ -868,6 +868,30 @@ fn extended_topology_info() {
...
+fn extended_topology_info_v2() {
+    use crate::TopologyType;
+
+    let cpuid = CpuId::with_cpuid_fn(cpuid_reader);
+    let mut e = cpuid
+        .get_extended_topology_info_v2()
+        .expect("Leaf is supported");
+
+    let t = e.next().expect("Have level 0");
+    assert_eq!(t.x2apic_id(), 199); // different from doc, unpinned execution
```
Comment (gz) [inline src/tests/xeon_gold_6252.rs:881]: I don't understand the comment (both what's different and what unpinned means here)?

File: src/tests/xeon_gold_6252.rs [hunk 2]
```diff
@@ -868,6 +868,30 @@ fn extended_topology_info() {
...
+    assert_eq!(t.level_type(), TopologyType::SMT);
+    assert_eq!(t.shift_right_for_next_apic_id(), 0x1);
+    assert_eq!(t.processors(), 2);
+
+    let t = e.next().expect("Have level 1");
+    assert_eq!(t.level_number(), 1);
+    assert_eq!(t.level_type(), TopologyType::Core);
+    assert_eq!(t.shift_right_for_next_apic_id(), 0x6);
+    assert_eq!(t.processors(), 48);
+    assert_eq!(t.x2apic_id(), 199); // different from doc, unpinned execution
```
Comment (gz) [inline src/tests/xeon_gold_6252.rs:892]: See above

File: src/lib.rs [hunk 3]
```diff
@@ -551,6 +552,9 @@ impl CpuId {
     }
 
     /// Information about topology (LEAF=0x0B).
+    /// 
+    /// Intel SDM suggests software should check support for leaf 0x1F ([`crate::get_extended_topology_info_v2()`]), and
```
Comment (gz) [inline src/lib.rs:556]: I think this should be [`CpuId::get_extended_topology_info_v2()`] instead?

Comment (semiviral) [reply src/lib.rs:556]: Agreed, I'll resolve this.

File: src/tests/xeon_gold_6252.rs (hunk 1 above)
Comment (semiviral) [reply src/tests/xeon_gold_6252.rs:881]: I'm really not sure—I coped the impl for the existing `extended_topology_info()` test, and only changed the CPUID call used within. Given that it seems like this test is pointless for this CPU (and the Ryzen Matisse), I'll just remove the test entirely.

**Comment by semiviral** [comment]
I'm currently running on an i7 12700k, which also supports the 0x1F leaf. I've gone ahead and added the CPU as a test model, as well as adding the (ostensibly new?) `LoadOnly` and `StoreOnly` values for `DatType`.

Additionally, I've gone ahead fixed the unsupported tests / leaves from the existing Xeon Gold and Ryzen Matisse CPU models.

**Comment by gz** [comment]
Awesome, thanks for all the work!

**Comment by gz** [comment]
Released as 10.4 on crates.io!

**Pull request merged**